File: 1929.py
import math
m, n = map(int, input().split())
num = [1 for i in range(n + 1)]
num[0] = num[1] = 0; lmt = int(math.sqrt(n))
    
for i in range(2, lmt + 1): 
    if num[i]:
        sum = i + i
        while sum <= n:
            num[sum] = 0
            sum += i
for i in range(m, n + 1):
    if num[i]: print(i) 

File: 1929.cpp
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main() {
    int m, n, sum, lmt;
	scanf("%d %d", &m, &n);
	int* num = (int*)calloc(n + 1, sizeof(int)); 
	num[0] = num[1] = 1; lmt = (int)sqrt(n);
	// 요소의 값이 0이면 소수 
	for (int i = 2; i <= lmt; i++) {
		if (num[i] == 0) {
			sum = i;
			while ((sum += i) <= n) 
				num[sum] = 1;
		}
	}
	for (; m <= n; m++)
		if (num[m] == 0) printf("%d\n", m); 
	free(num);
    return 0;
}
